client/src/components/shared/Button.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+
+/**
+ * Generic button component meant as a replacement for
+ * the button tag.
+ */
+const Button = styled.button.attrs((props) => ({
+  type: props.type || 'button',
+}))`
+  height: ${({ height }) => height};
+  width: ${({ width }) => width};
+  margin: 0;
+  background-color: transparent;
+  cursor: pointer;
+  ${({ variant }) => {
+    switch (variant) {
```

Comment:
Is there a reason to use a switch here and just an if statement?

Reply:
Not really. I guess I was thinking it would be more readable if we had many variants. We will likely want at least a secondary and nav variant at some point.

Reply:
Should probably just change it to if/else statements at some point. I'll make a ticket and leave it for later to try to make progress on some more urgent things first.